src/foam/box/socket/SocketServer.js

```diff
@@ -68,29 +73,42 @@ foam.CLASS({
           final ServerSocket serverSocket = serverSocket0;
 
           Agency agency = (Agency) getX().get(getThreadPoolName());
+          final X x_ = getX();
           logger.info("start", "threadPoolName", getThreadPoolName(), "threadsPerCore", ((foam.nanos.pool.AbstractFixedThreadPool) agency).getThreadsPerCore(), "numberOfThreads", ((foam.nanos.pool.AbstractFixedThreadPool) agency).getNumberOfThreads());
-          agency.submit(
-            getX(),
-            new ContextAgent() {
-              @Override
-              public void execute(X x) {
-                try {
-                  while ( true ) {
-                    Socket client = serverSocket.accept();
-                    client.setSoTimeout(getSoTimeout());
-                    agency.submit(
-                      x,
-                      new SocketServerProcessor(getX(), client),
-                      client.getRemoteSocketAddress().toString()
-                    );
-                  }
-                } catch ( IOException ioe ) {
-                  logger.error(ioe);
+          Thread serverThread = new Thread(() -> {
+            try {
+              while ( true ) {
+                Socket client = serverSocket.accept();
+                client.setSoTimeout(getSoTimeout());
+
+                if ( getUseThreadPoolForConnection() ) {
+                  agency.submit(
+                    x_,
+                    new SocketServerProcessor(x_, client),
+                    client.getRemoteSocketAddress().toString()
+                  );
+                } else {
+                  Thread acceptThread = new Thread(() -> {
+                    try {
+                      SocketServerProcessor processor = new SocketServerProcessor(x_, client);
+                      processor.execute(x_);
+                    } catch ( IOException ioe ) {
+                      logger.error("Failed to accecpt connection with: " + client.getRemoteSocketAddress(), ioe);
+                    }
+                  }, client.getRemoteSocketAddress().toString());
+                  acceptThread.setDaemon(true);
+                  acceptThread.start();
```

Comment:
SocketServerProcessor will need to call execute(x) on start() so that it can work as Thread directly and from threadpool which expects contextAgents.

Reply:
yes, it calls. see line 94